Mathlib/Topology/Homotopy/Lifting.lean

```diff
@@ -0,0 +1,365 @@
+/-
+Copyright (c) 2025 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.AlgebraicTopology.FundamentalGroupoid.SimplyConnected
+import Mathlib.Topology.Connected.LocPathConnected
+import Mathlib.Topology.Covering
+import Mathlib.Topology.Homotopy.Path
+import Mathlib.Topology.UnitInterval
+
+/-!
+# The homotopy lifting property for covering maps
+
+- `IsCoveringMap.exists_path_lifts`, `IsCoveringMap.liftPath`: any path in the base of a covering
+  map lifts uniquely to the covering space (given a lift of the starting point).
+
+- `IsCoveringMap.liftHomotopy`: any homotopy `I × A → X` in the base of a covering map `E → X` can
+  be lifted to a homotopy `I × A → E`, starting from a given lift of the restriction `{0} × A → X`.
+
+- `IsCoveringMap.existsUnique_continuousMap_lifts`: any continuous map from a simply-connected,
+  locally path-connected space lifts uniquely through a covering map (given a lift of an
+  arbitrary point).
+-/
+
+open Topology unitInterval
+
+variable {E X A : Type*} [TopologicalSpace E] [TopologicalSpace X] [TopologicalSpace A] {p : E → X}
+
+namespace IsLocalHomeomorph
+
+variable (homeo : IsLocalHomeomorph p)
+include homeo
+
+/-- If `p : E → X` is a local homeomorphism, and if `g : I × A → E` is a lift of `f : C(I × A, X)`
+  continuous on `{0} × A ∪ I × {a}` for some `a : A`, then there exists a neighborhood `N ∈ 𝓝 a`
+  and `g' : I × A → E` continuous on `I × N` that agrees with `g` on `{0} × A ∪ I × {a}`.
+  The proof follows [hatcher02], Proof of Theorem 1.7, p.30.
+
+  This lemma should also be true for an arbitrary space in place of `I` if `A` is locally connected
+  and `p` is a separated map, which guarantees uniqueness and therefore well-definedness
+  on the intersections. -/
+theorem exists_lift_nhds {f : C(I × A, X)} {g : I × A → E} (g_lifts : p ∘ g = f)
+    (cont_0 : Continuous (g ⟨0, ·⟩)) (a : A) (cont_a : Continuous (g ⟨·, a⟩)) :
+    ∃ N ∈ 𝓝 a, ∃ g' : I × A → E, ContinuousOn g' (Set.univ ×ˢ N) ∧ p ∘ g' = f ∧
+      (∀ a, g' (0, a) = g (0, a)) ∧ ∀ t, g' (t, a) = g (t, a) := by
+  -- For every `e : E`, upgrade `p` to a LocalHomeomorph `q e` around `e`.
+  choose q mem_source hpq using homeo
+  /- Using the hypothesis `cont_a`, we partition the unit interval so that for each subinterval
+   [tₙ, tₙ₊₁], g ([tₙ, tₙ₊₁] × {a}) is contained in the domain of some local homeomorphism `q e`. -/
```

Comment:
```suggestion
   `[tₙ, tₙ₊₁]`, `g ([tₙ, tₙ₊₁] × {a})` is contained in the domain of some local homeomorphism `q e`. -/
```

Reply:
I don't think this improves readability, since this isn't a docstring and doesn't get rendered.

Reply:
Well, I was confused for a second, thinking that it was one big expression.
In an ordinary LaTeX text, I would recommend adding some word between the two math expressions. Here the backticks are a compromise, but some filler words like
```suggestion
   [tₙ, tₙ₊₁], the image g ([tₙ, tₙ₊₁] × {a}) is contained in the domain of some local homeomorphism `q e`. -/
```
would also help.

Reply:
Okay, let me take both.